src/v0/destinations/klaviyo/util.js

```diff
@@ -1,116 +1,117 @@
+const { BASE_ENDPOINT, MAPPING_CONFIG, CONFIG_CATEGORIES, MAX_BATCH_SIZE } = require('./config');
+const { JSON_MIME_TYPE } = require('../../util/constant');
 const { NetworkError } = require('../../util/errorTypes');
+const { getDynamicErrorType } = require('../../../adapters/utils/networkUtils');
 const tags = require('../../util/tags');
-const { JSON_MIME_TYPE } = require('../../util/constant');
-
-/**
- * This function is used to check if the user/profile already exists or not, if already exists unique person_id
- * for that user is getting returned else false is returned.
- * Docs: https://developers.klaviyo.com/en/reference/get-profile-id
- * @param {*} message
- * @param {*} Config
- * @returns
- */
-const isProfileExist = async (message, { Config }) => {
-  const { privateApiKey } = Config;
-  const userIdentifiers = {
-    email: getFieldValueFromMessage(message, 'email'),
-    external_id: getFieldValueFromMessage(message, 'userId'),
-    phone_number: getFieldValueFromMessage(message, 'phone'),
-  };
+const { handleHttpRequest } = require('../../../adapters/network');
 
-  // eslint-disable-next-line no-restricted-syntax
-  for (const id in userIdentifiers) {
-    if (isDefinedAndNotNull(userIdentifiers[id])) {
-      // eslint-disable-next-line no-await-in-loop
-      const profileResponse = await httpGET(`${BASE_ENDPOINT}/api/v2/people/search`, {
-        header: {
-          Accept: JSON_MIME_TYPE,
-        },
-        params: {
-          api_key: privateApiKey,
-          [id]: userIdentifiers[id],
-        },
-      });
-      const processedProfileResponse = processAxiosResponse(profileResponse);
-      if (processedProfileResponse.status === 200 && processedProfileResponse.response?.id) {
-        return processedProfileResponse.response.id;
-      }
-      if (
-        !(
-          processedProfileResponse.status === 404 &&
-          processedProfileResponse?.response.detail ===
-            'There is no profile matching the given parameters.'
-        )
-      ) {
-        throw new NetworkError(
-          `The lookup call could not be completed with the error:
-          ${JSON.stringify(processedProfileResponse.response)}`,
-          processedProfileResponse.status,
-          {
-            [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedProfileResponse.status),
-          },
-          processedProfileResponse,
-        );
-      }
-    }
+const getProfileId = async (endpoint, payload, requestOptions) => {
+  let profileId;
+  const { httpResponse: resp } = await handleHttpRequest('post', endpoint, payload, requestOptions);
+  if (resp.response?.status === 201) {
+    profileId = resp.response?.data?.data?.id;
+  } else if (resp.response?.response?.status === 409) {
+    const { response } = resp.response;
+    profileId = response.data?.errors[0]?.meta?.duplicate_profile_id;
+  } else {
+    throw new NetworkError(
+      `Failed to create user due to ${resp.response?.data}`,
```

Comment:
Are we creating user here ? I see the method name as `getProfileId`

Reply:
Yes, we are creating the user and extracting the `profileId` from the response. If the user already exists, or if doesn't exist, we get `profileId` in both case but in a different format.